daprdocs/content/en/developing-applications/crewai/crewai-workflows.md

```diff
@@ -0,0 +1,204 @@
+@wfr.activity(name="run_task")
+def run_task_activity(ctx: WorkflowActivityContext, task_name: str):
+    print(f"Running CrewAI task: {task_name}", flush=True)
+
+    task = tasks[task_name]
+
+    # Create a Crew for just this one task
+    temp_crew = Crew(agents=[agent], tasks=[task])
+
+    # kickoff() works across CrewAI versions
+    result = temp_crew.kickoff()
+
+    return str(result)
+
+# ------------------------------------------------------------
+# 3. Workflow — orchestrates tasks durably
+# ------------------------------------------------------------
+@wfr.workflow(name="crewai_multi_task_workflow")
+def crewai_workflow(ctx: DaprWorkflowContext):
+    print("Starting multi-task CrewAI workflow", flush=True)
+
+    latest_news = yield ctx.call_activity(run_task_activity, input="latest_ai_news")
+    startup_summary = yield ctx.call_activity(run_task_activity, input="ai_startup_launches")
+    policy_updates = yield ctx.call_activity(run_task_activity, input="ai_policy_updates")
+
+    return {
+        "latest_news": latest_news,
+        "startup_summary": startup_summary,
+        "policy_updates": policy_updates,
+    }
+
+# ------------------------------------------------------------
+# 4. Runtime + Client (entry point)
+# ------------------------------------------------------------
+if __name__ == "__main__":
+    wfr.start()
+
+    client = DaprWorkflowClient()
+    instance_id = "crewai-multi-01"
+
+    client.schedule_new_workflow(
+        workflow=crewai_workflow,
+        input=None,
+        instance_id=instance_id
+    )
+
+    state = client.wait_for_workflow_completion(instance_id, timeout_in_seconds=60)
+    print(state.serialized_output)
+```
+
+### Create the Workflow Database Component
+
+Dapr Workflows persist durable state using any [Dapr state store]({{% ref supported-state-stores %}}) that supports workflows.
+Create a components directory, then create the file workflowstore.yaml:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: statestore
+spec:
+  type: state.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+  - name: actorStateStore
+    value: "true"
+```
+
+This component stores:
+
+* Checkpoints
+* Execution history
+* Deterministic resumption state
+* Final output data
+
+### Set a CrewAI LLM Provider
+
+CrewAI needs an LLM configuration or token to run. See instructions [here](https://docs.crewai.com/en/concepts/llms#setting-up-your-llm).
+
+### Run the Workflow
+
+Launch the CrewAI workflow using the Dapr CLI:
+
+```bash
+dapr run \
+  --app-id crewaiwf \
+  --dapr-grpc-port 50001 \
+  --resources-path ./components \
+  -- python3 ./crewai_workflow.py
+```
+
+As the workflow runs, each CrewAI task is executed as a durable activity.
+If the process crashes, the workflow resumes exactly where it left off.
+
+Open Zipkin to view workflow traces:
+
+```
+http://localhost:9411
+```
```

Comment:
I could not get this code to work, first due to the component discovery. Let's check the steps.